integration-tests/resteasy-reactive-kotlin/standard/src/main/kotlin/io/quarkus/it/resteasy/reactive/kotlin/ReactiveGreetingResource.kt

```diff
@@ -4,18 +4,23 @@ import kotlinx.coroutines.delay
 import java.io.BufferedReader
 import java.io.InputStreamReader
 import java.net.URL
+import javax.inject.Inject
 import javax.ws.rs.GET
 import javax.ws.rs.Path
 import javax.ws.rs.Produces
 import javax.ws.rs.core.MediaType
 
 @Path("/hello-resteasy-reactive")
-class ReactiveGreetingResource {
+class ReactiveGreetingResource @Inject constructor (val req : RequestScopedKotlinClass){
     @GET
     @Produces(MediaType.TEXT_PLAIN)
     suspend fun hello(): String {
+        req.message = "msg"
         delay(50)
         doSomeWork()
+        if (req.message != "msg") {
```

Comment:
I don't quite understand what this is testing, honestly.  do we expect `req` to get replaced some how?  otherwise it all looks pretty straightforward to me.

Reply:
It is making sure it resumed with the same request scope, it it was different the value would be lost.